rewrite-maven/src/main/java/org/openrewrite/maven/tree/ResolvedPom.java

```diff
@@ -424,6 +424,8 @@ private void resolveParentDependenciesRecursively(List<Pom> pomAncestry) throws
             for (Profile profile : pom.getProfiles()) {
                 if (profile.isActive(activeProfiles)) {
                     mergeDependencyManagement(profile.getDependencyManagement(), pomAncestry);
+                    // If two profiles are activated that both define the same dependency but with different scope/version
+                    // The first one now takes precedence, which is not how maven processes them, especially at runtime
```

Comment:
- Let's handle this in https://github.com/openrewrite/rewrite/pull/4270
```suggestion
```
Removing this comment then avoids conflicts there.